### PR TITLE
Initialize logging earlier in container startup sequence [run-systemtest]

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs @gjoranv @hmusum PR

### CODE REVIEW

**Review by hmusum — APPROVED** [review]

File: vespalog/src/main/java/com/yahoo/log/InitializeLog.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.yahoo.log;
+
+/**
+ * Sets up Vespa logging. Call a setup method to set up this.
+ *
+ * @author baldersheim
+ */
+public class InitializeLog {
```
Comment (gjoranv) [inline vespalog/src/main/java/com/yahoo/log/InitializeLog.java:8]: Was it intentional to put this class in a PublicApi package?

File: vespalog/src/main/java/com/yahoo/log/InitializeLog.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    static {
+        LogSetup.initVespaLogging("Container");
```
Comment (gjoranv) [inline vespalog/src/main/java/com/yahoo/log/InitializeLog.java:10]: The term "Container" seems a bit too specific for a generic module as vespalog.

File: jdisc_core/pom.xml [hunk 3]
```diff
@@ -167,6 +167,11 @@
             <version>${project.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>com.yahoo.vespa</groupId>
+            <artifactId>vespalog</artifactId>
```
Comment (gjoranv) [inline jdisc_core/pom.xml:172]: I would like to verify the effects this dependency has on e.g. the package exports from jdisc_core (classpath) and the set of dependencies made visible via container-dev, before this PR is merged.

File: vespalog/src/main/java/com/yahoo/log/InitializeLog.java (hunk 2 above)
Comment (baldersheim) [reply vespalog/src/main/java/com/yahoo/log/InitializeLog.java]: Agree, changed to 'unused-default'

File: vespalog/src/main/java/com/yahoo/log/InitializeLog.java (hunk 1 above)
Comment (baldersheim) [reply vespalog/src/main/java/com/yahoo/log/InitializeLog.java]: No, that is not necessary. I moved it to the log.impl and annotated that with ExportPackage.
I assume that is necessary to access it from other modules.

File: jdisc_core/pom.xml (hunk 3 above)
Comment (baldersheim) [reply jdisc_core/pom.xml:172]: OK, I will hold until green light.

**Review by hmusum — APPROVED** [review]

**Comment by baldersheim** [comment]
This seems to be a bit brittle, and there was already explicit support for this in jdisc.
So this is superceeded in #22456

**Pull request closed without merging**